test/Sdk.Generator.Tests/FunctionMetadataProviderGeneratorTests/NestedTypesTest.cs

```diff
@@ -0,0 +1,215 @@
+                    {
+                        public Task<ImmutableArray<IFunctionMetadata>> GetFunctionMetadataAsync(string directory)
+                        {
+                            var metadataList = new List<IFunctionMetadata>();
+                            var Function0RawBindings = new List<string>();
+                            Function0RawBindings.Add(@"{""name"":""req"",""type"":""HttpTrigger"",""direction"":""In"",""authLevel"":""User"",""methods"":[""get""]}");
+                            Function0RawBindings.Add(@"{""name"":""$return"",""type"":""http"",""direction"":""Out""}");
+
+                            var Function0 = new DefaultFunctionMetadata
+                            {
+                                Language = "dotnet-isolated",
+                                Name = "HttpTriggerSimple",
+                                EntryPoint = "MyCompany.MyProject.HttpTriggerSimple.Run",
+                                RawBindings = Function0RawBindings,
+                                ScriptFile = "TestProject.dll"
+                            };
+                            metadataList.Add(Function0);
+
+                            return Task.FromResult(metadataList.ToImmutableArray());
+                        }
+                    }
+
+                    public static class WorkerHostBuilderFunctionMetadataProviderExtension
+                    {
+                        ///<summary>
+                        /// Adds the GeneratedFunctionMetadataProvider to the service collection.
+                        /// During initialization, the worker will return generated function metadata instead of relying on the Azure Functions host for function indexing.
+                        ///</summary>
+                        public static IHostBuilder ConfigureGeneratedFunctionMetadataProvider(this IHostBuilder builder)
+                        {
+                            builder.ConfigureServices(s => 
+                            {
+                                s.AddSingleton<IFunctionMetadataProvider, GeneratedFunctionMetadataProvider>();
+                            });
+                            return builder;
+                        }
+                    }
+                }
+                """;
+
+                await TestHelpers.RunTestAsync<FunctionMetadataProviderGenerator>(
+                    _referencedExtensionAssemblies,
+                    inputCode,
+                    expectedGeneratedFileName,
+                    expectedOutput);
+            }
+            [Fact]
+            public async Task NestedClass()
+            {
+                string inputCode = """
+                using System;
+                using System.Collections.Generic;
+                using Microsoft.Azure.Functions.Worker;
+                using Microsoft.Azure.Functions.Worker.Http;
+
+                namespace MyCompany
+                {
+                    public class  WebApis
```

Comment:
The legacy metadata generator fails to generate for this case (nested classes).

Reply:
Do we want to fail to generate in this case? Not sure what the pros/cons are, but I see the test is allowing this behavior.

Reply:
It's probably a good thing to supported nested classes

Reply:
> Do we want to fail to generate in this case? Not sure what the pros/cons are, but I see the test is allowing this behavior.

No. The legacy generator behavior is wrong here. It should still generate.